ibc-apps/ics20-transfer/types/src/error.rs

```diff
@@ -17,86 +16,46 @@ pub enum TokenTransferError {
-    InsufficientFunds {
-        send_attempt: String,
-        available_funds: String,
-    },
-    /// destination channel not found in the counterparty of port_id `{port_id}` and channel_id `{channel_id}`
-    DestinationChannelNotFound {
-        port_id: PortId,
-        channel_id: ChannelId,
-    },
-    /// base denomination is empty
-    EmptyBaseDenom,
-    /// invalid prot id n trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTracePortId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// invalid channel id in trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTraceChannelId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// malformed trace: `{0}`
-    MalformedTrace(String),
-    /// trace length must be even but got: `{len}`
-    InvalidTraceLength { len: u64 },
+    /// invalid trace: `{0}`
+    InvalidTrace(String),
     /// invalid amount: `{0}`
     InvalidAmount(FromDecStrErr),
     /// invalid token
     InvalidToken,
-    /// expected `{expect_order}` channel, got `{got_order}`
-    ChannelNotUnordered {
-        expect_order: Order,
-        got_order: Order,
+    /// invalid coin: `{actual}`
+    InvalidCoin { actual: String },
+    /// missing destination channel `{channel_id}` on port `{port_id}`
+    MissingDestinationChannel {
+        port_id: PortId,
+        channel_id: ChannelId,
     },
-    /// channel cannot be closed
-    CantCloseChannel,
+    /// mismatched channel orders: expected `{expected}`, actual `{actual}`
+    MismatchedChannelOrders { expected: Order, actual: Order },
+    /// mismatched port IDs: expected `{expected}`, actual `{actual}`
+    MismatchedPortIds { expected: PortId, actual: PortId },
     /// failed to deserialize packet data
-    PacketDataDeserialization,
+    FailedToDeserializePacket,
     /// failed to deserialize acknowledgement
-    AckDeserialization,
-    /// receive is not enabled
-    ReceiveDisabled { reason: String },
-    /// send is not enabled
-    SendDisabled { reason: String },
-    /// failed to parse as AccountId
-    ParseAccountFailure,
-    /// invalid port: `{port_id}`, expected `{exp_port_id}`
-    InvalidPort {
-        port_id: PortId,
-        exp_port_id: PortId,
-    },
-    /// decoding raw msg error: `{reason}`
-    DecodeRawMsg { reason: String },
-    /// unknown msg type: `{msg_type}`
-    UnknownMsgType { msg_type: String },
-    /// invalid coin string: `{coin}`
-    InvalidCoin { coin: String },
-    /// decoding raw bytes as UTF-8 string error: `{0}`
-    Utf8Decode(Utf8Error),
-    /// other error: `{0}`
-    Other(String),
+    FailedToDeserializeAck,
+    /// failed to parse account ID
+    FailedToParseAccount,
+    /// failed to decode raw msg: `{description}`
+    FailedToDecodeRawMsg { description: String },
+    /// channel cannot be closed
+    ClosedChannel,
```

Comment:
Give the usage, looks like it should be: `UnsupportedCloseChannel`

Reply:
[250766f](https://github.com/cosmos/ibc-rs/pull/1318/commits/250766f61141945b47b24c7b04cd6d7c82fb07b6)